.github/workflows/publish_snapshot.yml

```diff
@@ -30,11 +37,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        branch:
-          - master
-          - branch-3.5
-          - branch-3.4
-          - branch-3.3
+        # keep in sync with default value of workflow_dispatch input 'branch'
+        branch: ${{ fromJSON( inputs.branch || '["master", "branch-3.5", "branch-3.4", "branch-3.3"]' ) }}
```

Comment:
Hm, I don't think this helps. individual job within matrix has a isolated docker container to run the CI